src/Controller/DiscordController.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Slack Secret Santa project.
+ *
+ * (c) JoliCode <coucou@jolicode.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Joli\SlackSecretSanta\Controller;
+
+use Joli\SlackSecretSanta\Application\DiscordApplication;
+use Joli\SlackSecretSanta\User;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+use Wohali\OAuth2\Client\Provider\Discord;
+
+class DiscordController extends AbstractController
+{
+    private $discordClientId;
+    private $discordClientSecret;
+    private $router;
+
+    public function __construct(string $discordClientId, string $discordClientSecret, RouterInterface $router)
+    {
+        $this->router = $router;
+        $this->discordClientId = $discordClientId;
+        $this->discordClientSecret = $discordClientSecret;
+    }
+
+    /**
+     * Ask for Discord authentication and store the AccessToken in Session.
+     */
+    public function authenticate(Request $request, DiscordApplication $discordApplication): Response
+    {
+        $session = $request->getSession();
+
+        $provider = new Discord([
+            'clientId' => $this->discordClientId,
+            'clientSecret' => $this->discordClientSecret,
+            'redirectUri' => $this->router->generate('discord_authenticate', [], RouterInterface::ABSOLUTE_URL),
```

Comment:
Why not setting this object as a service?

Reply:
Because the `redirectUri` option is required in the constructor and I'm not sure it's worth the YAML pain to configure a service and generate an url inside the DIC (if that's even possible)